bin/vyper

```diff
@@ -13,28 +13,46 @@ sys.tracebacklimit = 0
 
 parser = argparse.ArgumentParser(description='Vyper {0} programming language for Ethereum'.format(vyper.__version__))
 parser.add_argument('input_file', help='Vyper sourcecode to compile')
-parser.add_argument('-f', help='Format to print', choices=['abi', 'json', 'bytecode', 'bytecode_runtime', 'ir', 'asm'], default='bytecode', dest='format')
+parser.add_argument('-f', help='Format to print', choices=['abi', 'json', 'bytecode', 'bytecode_runtime', 'ir', 'asm'], default=['bytecode'], dest='format', nargs='+')
 parser.add_argument('--show-gas-estimates', help='Show gas estimates in ir output mode.', action="store_true")
 
 args = parser.parse_args()
 
 
-def print_asm(asm_list):
+def get_asm(asm_list):
+    output_string = ''
     skip_newlines = 0
     for node in asm_list:
         if isinstance(node, list):
-            print_asm(node)
+            output_string += get_asm(node)
             continue
 
         is_push = isinstance(node, str) and node.startswith('PUSH')
 
-        print(str(node) + ' ', end='')
+        output_string += str(node) + ' '
         if skip_newlines:
             skip_newlines -= 1
         elif is_push:
             skip_newlines = int(node[4:]) - 1
         else:
-            print('')
+            output_string += '\n'
+    return output_string
+
+
+def format_output(code, return_format):
```

Comment:
Architecture suggestion:
```python
format_output = {}  # Makes a dict lookup of functions to apply
format_output['abi'] = lambda code: compiler.mk_full_signature(code)
format_output['json'] = lambda code: json.dumps(compiler.mk_full_signature(code))
# Add the rest...

# Now we don't have to modify this to add new output types
parser.add_argument('-f', choices=format_output.keys(), ...)
# Other argparse stuff...

for i in args.format:
    print(format_output[i](code))  # Should be impossible to get a key error because of argparse
```
---

This has the benefit of DRY of course, but also you can be a little bit more selective in application of which keys conflict easier e.g.
```python
# Logical NAND of output formats for ABI
assert not ('abi' in args.format) or not ('json' in args.format), \
        "Only Python or JSON output allowed for ABI!"
```

Reply:
Good suggestion, I've made the change.

I'm not sure if there should be a restriction on the output formats that are allowed. If for some crazy reason someone wants both the abi & json, why not display both?